tests/amd_migraphx_test/runMIGraphXTests/runMIGraphXTests.cpp

```diff
@@ -0,0 +1,735 @@
+            parameter++;
+        }
+        else if (!strcasecmp(argv[arg], "--squeezenet")) {
+            if ((arg + 1) == argc) {
+                printf("\n\nERROR: missing squeezenet ONNX .model file location on command-line (see help for details)\n\n\n");
+                show_usage();
+                exit(-1);
+            }
+            runSqueezenet = true;
+            runAnyImagenet = true;
+            arg++;
+            binaryFilename_squeezenet_str = (argv[arg]);
+            parameter++;
+        }
+        else if (!strcasecmp(argv[arg], "--resnet50")) {
+            if ((arg + 1) == argc) {
+                printf("\n\nERROR: missing resnet50 ONNX .model file location on command-line (see help for details)\n\n\n");
+                show_usage();
+                exit(-1);
+            }
+            runResnet50 = true;
+            runAnyImagenet = true;
+            arg++;
+            binaryFilename_resnet50_str = (argv[arg]);
+            parameter++;
+        }
+
+        else if (!strcasecmp(argv[arg], "--vgg19")) {
+            if ((arg + 1) == argc) {
+                printf("\n\nERROR: missing vgg19 ONNX .model file location on command-line (see help for details)\n\n\n");
+                show_usage();
+                exit(-1);
+            }
+            runVgg19 = true;
+            runAnyImagenet = true;
+            arg++;
+            binaryFilename_vgg19_str = (argv[arg]);
+            parameter++;
+        }
+        else if (!strcasecmp(argv[arg], "--googlenet")) {
+            if ((arg + 1) == argc) {
+                printf("\n\nERROR: missing googlenet ONNX .model file location on command-line (see help for details)\n\n\n");
+                show_usage();
+                exit(-1);
+            }
+            runGooglenet = true;
+            runAnyImagenet = true;
+            arg++;
+            binaryFilename_googlenet_str = (argv[arg]);
+            parameter++;
+        }
+        else if (!strcasecmp(argv[arg], "--densenet")) {
+            if ((arg + 1) == argc) {
+                printf("\n\nERROR: missing densenet ONNX .model file location on command-line (see help for details)\n\n\n");
+                show_usage();
+                exit(-1);
+            }
+            runDensenet = true;
+            runAnyImagenet = true;
+            arg++;
+            binaryFilename_densenet_str = (argv[arg]);
+            parameter++;
+        }
+    }
+
+    if (parameter < 2) {
+        printf("\nERROR: missing parameters in command-line.\n");
+        show_usage();
+        exit(-1);
+    }
+
+    // create context, input, output, and graph
+    vx_status status = 0;
+    vx_context context = vxCreateContext();
+    ERROR_CHECK_OBJECT(context);
+    vxRegisterLogCallback(context, log_callback, vx_false_e);
+    
+    // load vx_nn kernels
+    ERROR_CHECK_STATUS(vxLoadKernels(context, "vx_amd_migraphx"));
+
+    // creation of graphs
+    vx_graph graph_mnist = vxCreateGraph(context);
```

Comment:
please create graph_mnist only if runMnist or runAnyImagenet are true. use similar logic for other models too.